### PR TITLE
Bug/first time routing

### PR DESCRIPTION
**Pull request by Daniela510**

### CODE REVIEW

File: lib/features/basic_giving_flow/pages/donation_success.dart [hunk 1]
```diff
@@ -14,6 +14,10 @@ class SuccessDonationPage extends StatelessWidget {
 
   final NavigationService _navigationService = locator<NavigationService>();
   late final LocalStorageProxy storageProxy = locator<LocalStorageProxy>();
```
Comment (AlexGivt) [inline lib/features/basic_giving_flow/pages/donation_success.dart:16]: I would have expected an "interface" or something similar here, not a specific store thing "LocalStorageProxy"

File: lib/features/basic_giving_flow/pages/donation_success.dart [hunk 2]
```diff
@@ -14,6 +14,10 @@ class SuccessDonationPage extends StatelessWidget {
...
+  @override
+  void initState() {
+    storageProxy.updateProgress('completedOneDonation');
```
Comment (AlexGivt) [inline lib/features/basic_giving_flow/pages/donation_success.dart:19]: No magic strings, please. I would prefer methods on the proxy, but if that is too difficult, gather the keys in a class or something and reference them by name

File: lib/features/home_screen/pages/first_use.dart [hunk 3]
```diff
@@ -20,13 +20,17 @@ class FirstUsePage extends StatefulWidget {
 class _FirstUsePageState extends State<FirstUsePage> {
   CheckInternet? _checkInternet;
   final NavigationService _navigationService = locator<NavigationService>();
+  late final LocalStorageProxy storageProxy = locator<LocalStorageProxy>();
```
Comment (AlexGivt) [inline lib/features/home_screen/pages/first_use.dart:23]: Inject and use interface, not actual class

File: lib/features/home_screen/pages/first_use.dart [hunk 4]
```diff
@@ -20,13 +20,17 @@ class FirstUsePage extends StatefulWidget {
...
 
   @override
   void initState() {
     _checkInternet = Provider.of<CheckInternet>(context, listen: false);
     _checkInternet?.checkRealtimeConnection();
     initialization();
     super.initState();
+    final LocalStorage current = storageProxy.realm.all<LocalStorage>().first;
```
Comment (AlexGivt) [inline lib/features/home_screen/pages/first_use.dart:31]: ??? what structure does this follow? Seems out of place to have a hard link to realm here

File: lib/features/basic_giving_flow/pages/donation_success.dart (hunk 2 above)
Comment (phalox) [reply lib/features/basic_giving_flow/pages/donation_success.dart:19]: Please use an ENUM for this: https://www.tutorialspoint.com/dart_programming/dart_programming_enumeration.htm

File: lib/models/localStorage.dart [hunk 5]
```diff
@@ -95,25 +94,25 @@ class LocalStorageProxy {
   void updateProgress(String key) {
     realm.write(() {
       LocalStorage localCurrent = realm.all<LocalStorage>().first;
-      if (key == 'camera') {
-        localCurrent.cameraAsked = true;
+      if (key == 'welcomed') {
```
[the remark this answers is not in the document]
Comment (phalox) [reply lib/models/localStorage.dart:97]: yeah, never mind about enum. I'd implement a dedicated method for this for now.

Also, they don't REALLY feel like a progress bar. so I'd just keep them separate

setWelcomedFlag(true)

setCompletedOneDonationFlag(true)

File: lib/utils/router.dart [hunk 6]
```diff
@@ -24,9 +23,7 @@ Route<dynamic> generateRoute(RouteSettings settings) {
     case routes.LoginRoute:
       final String? email = settings.arguments as String?;
       return MaterialPageRoute(
-          builder: (context) =>  LoginPage(passedEmail: email));
-    case routes.UspRoute:
```
Comment (phalox) [inline lib/utils/router.dart:28]: What happened to this page ?

**Review by phalox — COMMENTED** [review]
More thoughts added

File: lib/models/localStorage.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (AlexGivt) [reply lib/models/localStorage.dart:97]: Besides this, calculating/showing progress is another concern, separate this from storage

File: lib/utils/router.dart (hunk 6 above)
Comment (Daniela510) [reply lib/utils/router.dart:28]: it was from the first iteration of the flow, it doesn't exist in this flow and wont come back.

File: lib/models/localStorage.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/models/localStorage.dart]: These flags are meant to inform what flow the user should enter, the one with the intro animation& permissions, just the first use screen, or have they completed a donation and are directed to the home screen. This realms storage is meant to save state persistent between app lifecycles. It is the place we decided to store this information(flags).

File: lib/features/permissions/pages/camera_permission.dart [hunk 7]
```diff
@@ -29,7 +29,7 @@ class _CameraPermissionPageState extends State<CameraPermissionPage>
   void initState() {
     super.initState();
     WidgetsBinding.instance.addObserver(this);
-    _model.updateProgress('camera');
+    _storageProxy.setWelcomeFlag(true);
```
Comment (AlexGivt) [inline lib/features/permissions/pages/camera_permission.dart:32]: Seems weird, from 'camera' to Welcome?

Comment (Daniela510) [reply lib/features/permissions/pages/camera_permission.dart:32]: Yes, camera permissions page is the last screen in the welcoming sequence of screens. So if a user reached that page the flag is updated.

**Review by AlexGivt — APPROVED** [review]
Further refactoring can be done in story 399

**Pull request merged**